Contributions toward fixing #22

JJ opened the pull request:
Still incomplete, but at least narrows the search.

JJ committed:
Making some additional tests

JJ committed:
Adds more tests

But mainly tries to address #22 pinpointing the origin of the problem, which is that @globbers value is nil. This is why I have added some tests, and in order to do that, exported simplify, which helps in that.

I think that the problem is somewhere along the lines of creatting globbers, but without extensive testing of the grammars and base classes, it's difficult to dind it.

zostay commented:
When exporting `simpify` for testing, I would be okay with something like `is export(:TESTING)` or something. I do not want to pollute the callers namespace by default.

JJ committed:
Export selectively simplify

And also return the next unless line to where it was. It does not really fix absolute.t, and it breaks dir.t. So it's better if it stays.

JJ commented:
@zostay done, and also reverted the change that fixed two fails in absolute.t creating a whole lot of them in dir.t.

JJ committed:
:see_no_evil:

JJ committed:
Done testing when it needs to.

JJ committed:
Fixed first subtest. Refs #22

JJ committed:
Change test, closes #22

absolute.t and dir.t are incompatible. This is the only way they can both work at the same time. This fixes the inmediate problem, if that's not according ot spec more tests should be created.

JJ committed:
Works fine with latest commit.

lizmat commented:
I think this PR can now be closed?

JJ commented:
Well, it still includes new tests that didn't exist before.

JJ commented:
Also, there was an unrelated error that was fixed.

JJ committed:
Eliminates debug code

JJ commented:
Ping?
